### PR TITLE
[DX] Inform about deprecated rules in config

### PR DESCRIPTION
**Pull request by TomasVotruba**
Let's group reports together in the end, so they're noticed and can be resolved one by one:

Screenshot from 2024-06-26 15-30-51

### CODE REVIEW

File: src/Console/Command/ProcessCommand.php [hunk 1]
```diff
@@ -91,6 +93,9 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
         $outputFormatter->report($processResult, $configuration);
 
+        $this->deprecatedRulesReporter->reportDeprecatedRules();
```
Comment (samsonasik) [inline src/Console/Command/ProcessCommand.php:98]: I think this should be shown before any rule executed, which on ApplicationFileProcessor, at 

https://github.com/rectorphp/rector-src/blob/5bb9cbafb58949e3e05c5c3a294d86df2c4902a0/src/Application/ApplicationFileProcessor.php#L60-L61

On that part, only shown once

Comment (TomasVotruba) [reply src/Console/Command/ProcessCommand.php:98]: The idea to show errors in the end is to make it easier to spot.
If we run Rector, then show some errors, then make progress, in the end it "looks good".

If we put these notifications in the end, they won't be missed.

File: rules/CodingStyle/Rector/Foreach_/MultiDimensionalArrayToArrayDestructRector.php [hunk 2]
```diff
@@ -152,7 +152,7 @@ private function getUsedVariableNamesInForeachTree(Foreach_ $foreach): array
         $variableNodes = $this->nodeFinder->findInstanceOf($foreach, Variable::class);
 
         return array_unique(array_map(
-            fn (Variable $variable): string => (string) $this->getName($variable),
+            fn (Variable $variable): string => (string) $this->getName($foreach),
```
Comment (samsonasik) [inline rules/CodingStyle/Rector/Foreach_/MultiDimensionalArrayToArrayDestructRector.php]: this is bug on `RenameParamToMatchTypeRector`,  that cause test error, need use `$variable`:

```suggestion
            fn (Variable $variable): string => (string) $this->getName($variable),
```


Ref issue and PR:

- https://github.com/rectorphp/rector/issues/8693
- https://github.com/rectorphp/rector-src/pull/6046

Comment (TomasVotruba) [reply rules/CodingStyle/Rector/Foreach_/MultiDimensionalArrayToArrayDestructRector.php]: Got it :+1:

File: .github/workflows/e2e_consecutive_changes.yaml [hunk 3]
```diff
@@ -1,71 +0,0 @@
...
-
-            # run e2e test with dry run, we expect non-zero exit code
-            -   run: php ../e2eTestChangingRunnerWithCache.php -o expected-output-1-dry-run.diff --dry-run
-                working-directory: ${{ matrix.directory }}
-                continue-on-error: true
-                env:
-                    RECTOR_DISABLE_PARALLEL: ${{ matrix.rector_disable_parallel }}
-
-            # run e2e test with dry run once more, we expect non-zero exit code again
-            -   run: php ../e2eTestChangingRunnerWithCache.php -o expected-output-2-dry-run.diff --dry-run
```
Comment (samsonasik) [inline .github/workflows/e2e_consecutive_changes.yaml:55]: `e2e/e2eTestChangingRunnerWithCache.php` can be removed since no longer used.

**Comment by sanfair** [comment]
@TomasVotruba will this PR remove warnings for PostRector updates?

I use it with [NameImportingPostRector](https://github.com/rectorphp/rector/blob/6be87806c20a4ee5929742a349ed3f823dd71b52/src/PostRector/Rector/NameImportingPostRector.php#L29):
```php
use Rector\Config\RectorConfig;

return RectorConfig::configure()
    ->withSkip([
        NameImportingPostRector::class => [
            __DIR__.'/bootstrap/providers.php',
            __DIR__.'/config',
        ],
    ])
```

image

**Comment by TomasVotruba** [comment]
@sanfair  Good point, I'll add exception for post rules. Thanks :+1:

**Pull request merged**